### ISSUE TITLE
Button group overflow should be an aria menu

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3515]
## Describe the bug

The Button Group's Overflow feature should be implemented as an ARIA menu.

The current Button Group Overflow feature has 2 issues:
1) The overflow button does not have an accessible name
2) The overflow items do not support arrow key navigation instead of each item being in the tab order. This could be severe if there are a large number of overflow items.
3) The focus indicator on overflow items is suppressed

## How to reproduce

https://clarity.design/documentation/button-group
- Locate the Overflow example
- Using a screen reader, attempt to read the Overflow button. Observe that you do not hear a name for this button
- Turn the screen reader off
- Observe that you cannot choose "Download" or "Delete" using the arrow keys and Enter/Space as users may expect when presented with a menu.
- Notice the focus indicator is missing

## Expected behavior

Implement the Overflow as an ARIA menu like the clr-dropdown-menu in the header component or just use the clr-dropdown-menu component instead.

OR

Provide an accessible name for the Overflow button using aria-label. Since the content is inline with the button we can use aria-expanded on the button to indicate whether the overflow items are showing or not.

AND

Ensure the focus indicator is not suppressed (which may be fixed by the above option 1).

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

Be wary of breaking changes with these modifications.
